File: frontend/src/models/customer_credit.rs
//! 客户信用模型

use serde::{Deserialize, Serialize};

/// 客户信用数据模型
#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct CustomerCredit {
    pub id: i32,
    pub customer_id: i32,
    pub credit_level: Option<String>,
    pub credit_score: Option<i32>,
    pub credit_limit: Option<String>,
    pub used_credit: Option<String>,
    pub available_credit: Option<String>,
    pub credit_days: Option<i32>,
    pub status: Option<String>,
    pub remark: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

/// 客户信用列表响应
#[derive(Debug, Clone, Deserialize)]
pub struct CustomerCreditListResponse {
    pub items: Vec<CustomerCredit>,
    pub total: Option<u64>,
    pub page: u64,
    pub page_size: u64,
}

/// 客户信用查询参数
#[derive(Debug, Clone, Serialize)]
pub struct CreditQueryParams {
    pub customer_id: Option<i32>,
    pub credit_level: Option<String>,
    pub status: Option<String>,
    pub page: Option<i64>,
    pub page_size: Option<i64>,
}

/// 信用评级设置请求
#[derive(Debug, Clone, Serialize)]
pub struct CreditRatingRequest {
    pub customer_id: i32,
    pub credit_level: String,
    pub credit_score: i32,
    pub credit_limit: String,
    pub credit_days: i32,
    pub remark: Option<String>,
}

/// 信用额度调整请求
#[derive(Debug, Clone, Serialize)]
pub struct CreditLimitAdjustmentRequest {
    pub adjustment_type: String,
    pub amount: String,
    pub reason: String,
}

/// 信用额度占用/释放请求
#[derive(Debug, Clone, Serialize)]
pub struct CreditAmountRequest {
    pub amount: String,
}

File: frontend/src/api/customerCredit.ts
import { request } from './request'

export interface QueryParams {
  page?: number
  pageSize?: number
  customerId?: number
  creditLevel?: string
  status?: string
}

export interface CustomerCredit {
  id?: number
  customerId?: number
  customerName?: string
  creditLevel?: string
  creditScore?: number
  creditLimit?: number
  creditDays?: number
  usedAmount?: number
  availableAmount?: number
  status?: string
  remark?: string
  createdAt?: string
  updatedAt?: string
}

export interface CreditRatingRequest {
  customerId: number
  creditLevel: string
  creditScore: number
  creditLimit: number
  creditDays: number
  remark?: string
}

export interface CreditLimitAdjustmentRequest {
  adjustmentType: string
  amount: number
  reason: string
}

export interface CreditAmountRequest {
  amount: number
}

export function listCredits(params?: QueryParams) {
  return request.get('/api/v1/customer-credit', { params })
}

export function getCredit(customerId: number) {
  return request.get(`/api/v1/customer-credit/${customerId}`)
}

export function createCredit(data: Partial<CustomerCredit>) {
  return request.post('/api/v1/customer-credit', data)
}

export function updateCredit(id: number, data: Partial<CustomerCredit>) {
  return request.put(`/api/v1/customer-credit/${id}`, data)
}

export function deleteCredit(id: number) {
  return request.delete(`/api/v1/customer-credit/${id}`)
}

export function setCreditRating(data: CreditRatingRequest) {
  return request.post('/api/v1/customer-credit/set-rating', data)
}

export function occupyCredit(customerId: number, data: CreditAmountRequest) {
  return request.post(`/api/v1/customer-credit/${customerId}/occupy`, data)
}

export function releaseCredit(customerId: number, data: CreditAmountRequest) {
  return request.post(`/api/v1/customer-credit/${customerId}/release`, data)
}

export function adjustCreditLimit(customerId: number, data: CreditLimitAdjustmentRequest) {
  return request.post(`/api/v1/customer-credit/${customerId}/adjust`, data)
}

export function deactivateCredit(customerId: number) {
  return request.post(`/api/v1/customer-credit/${customerId}/deactivate`)
}
